### PR TITLE
Verify CRD's condition to ensure it's registered with k8s API (rebased)

### PR DESCRIPTION
**Pull request by jpeeler**
This is Vu's https://github.com/operator-framework/operator-lifecycle-manager/pull/553, but rebased and with tests passing (should be at least!). It has been modified from his original code slightly - ~mainly instead of creating a crdWithStatus, instead uses crdWithoutStatus which required far less code modification.~

### CODE REVIEW

**Comment by jpeeler** [comment]
Copied from old PR:

OLM needs to verify CRD's condition to make sure it is registered
with k8s API during CSV control loop before installing operator.
This verification will help to avoid errors related to the
unavailability of an API when installing an operator via OLM.

Jira: https://jira.coreos.com/browse/ALM-775

File: pkg/controller/operators/olm/requirements_test.go [hunk 1]
```diff
@@ -453,6 +474,58 @@ func TestRequirementAndPermissionStatus(t *testing.T) {
...
+			},
+			expectedError: nil,
+		},
+		{
+			description: "RequirementNotMet/NamesConflictedCRDVersion",
+			csv: csv("csv1",
+				namespace,
+				"",
+				installStrategy("csv1-dep", nil, nil),
+				[]*v1beta1.CustomResourceDefinition{crdWithoutStatus("c1", "v2")},
```
Comment (ecordell) [inline pkg/controller/operators/olm/requirements_test.go]: I think I'm missing what makes this CRD be unavailable? What's the difference between this test and the previous?

File: pkg/controller/operators/olm/requirements.go [hunk 2]
```diff
@@ -3,11 +3,13 @@ package olm
 import (
 	"encoding/json"
 	"fmt"
+
 	"github.com/sirupsen/logrus"
 
 	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
 	olmErrors "github.com/operator-framework/operator-lifecycle-manager/pkg/controller/errors"
 	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/install"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
```
Comment (njhale) [inline pkg/controller/operators/olm/requirements.go]: nit: import ordering

Comment (jpeeler) [reply pkg/controller/operators/olm/requirements.go]: I thought my tooling was out of date, but I just did a `go get golang.org/x/tools/cmd/goimports` and ran goimports on requirements.go and it still put the import line in the same spot. Where does yours get put?

**Comment by jpeeler** [comment]
/retest
This passed for me locally.

File: pkg/controller/operators/olm/requirements.go [hunk 3]
```diff
@@ -28,36 +30,59 @@ func (a *Operator) requirementStatus(strategyDetailsDeployment *install.Strategy
...
+					if version.Served {
+						status.Status = v1alpha1.RequirementStatusReasonPresent
+						served = true
+					}
+					break
+				}
+			}
+
+			if !served {
+				status.Status = v1alpha1.RequirementStatusReasonNotPresent
```
[the remark this answers is not in the document]
Comment (chancez) [reply pkg/controller/operators/olm/requirements.go]: Oh god please yes. This is the worst part of debugging these things.

**Pull request merged**